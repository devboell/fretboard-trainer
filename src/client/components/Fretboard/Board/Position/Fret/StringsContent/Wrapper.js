import styled from 'styled-components'

const background = ({ theme, isHighlighted }) => (
  isHighlighted
    ? theme.highlight
    : theme.background
)

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 5%;
  background-color: ${props => background(props)};
`

export default Wrapper
