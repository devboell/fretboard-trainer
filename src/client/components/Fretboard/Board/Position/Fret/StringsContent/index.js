import React from 'react'
import pt from 'prop-types'
import Wrapper from './Wrapper'


const StringsContent = ({ note, isHighlighted }) =>
  <Wrapper isHighlighted={isHighlighted}>
    {note}
  </Wrapper>

StringsContent.propTypes = {
  note: pt.string.isRequired,
  isHighlighted: pt.bool.isRequired,
}

export default StringsContent
