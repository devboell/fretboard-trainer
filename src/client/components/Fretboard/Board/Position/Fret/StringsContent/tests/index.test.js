import React from 'react'

import StringsContent from '../index'


const props = {
  note: 'E2',
  isHighlighted: true,
}

it('StringsContent component, snapshot', () => {
  const wrapper = shallow(<StringsContent {...props} />)
  expect(wrapper).toMatchSnapshot()
})
