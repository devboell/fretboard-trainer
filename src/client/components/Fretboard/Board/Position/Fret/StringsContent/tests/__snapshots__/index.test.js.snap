// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StringsContent component, snapshot 1`] = `
<styled.div
  isHighlighted={true}
>
  E2
</styled.div>
`;
